<launch>
   <arg name="odom_frame_id"   default="odom"/>
   <arg name="base_frame_id"   default="base_link"/>
   <arg name="global_frame_id" default="map"/>
   <arg name="odom_topic" default="/pepper_robot/odom" />
   <arg name="laser_topic" default="/pepper_robot/laser" />
   <arg name="map_topic"  default="/rtabmap/grid_map"/>
   <arg name="cmd_topic"   default="/cmd_vel"/>
   <arg name="gazebo"/>
   <!-- if $(arg gazebo) is false do this unles another-->
   <group unless="$(arg gazebo)">
      <node pkg="move_base" type="move_base" respawn="false" name="move_base" >
         <rosparam file="$(find movebase_pepper)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
         <rosparam file="$(find movebase_pepper)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
         <rosparam file="$(find movebase_pepper)/params/local_costmap_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/global_costmap_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/dwa_local_planner_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/move_base_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/global_planner_params.yaml" command="load"/>
         <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
         <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
         <param name="local_costmap/global_frame" value="$(arg global_frame_id)"/>
         <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
         <param name="DWAPlannerROS/global_frame_id" value="$(arg global_frame_id)"/>
         <remap from="cmd_vel" to="$(arg cmd_topic)"/>
         <remap from="odom" to="$(arg odom_topic)"/>
         <remap from="scan" to="$(arg laser_topic)"/>
         <remap from="map"  to="$(arg map_topic)"/>
      </node>
   </group>
   <!-- if $(arg gazebo) is true do this -->
   <group if="$(arg gazebo)">
      <node pkg="move_base" type="move_base" respawn="false" name="move_base" >
         <rosparam file="$(find movebase_pepper)/params/gazebo/costmap_common_params.yaml" command="load" ns="global_costmap" />
         <rosparam file="$(find movebase_pepper)/params/gazebo/costmap_common_params.yaml" command="load" ns="local_costmap" />
         <rosparam file="$(find movebase_pepper)/params/gazebo/local_costmap_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/gazebo/global_costmap_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/gazebo/dwa_local_planner_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/gazebo/move_base_params.yaml" command="load" />
         <rosparam file="$(find movebase_pepper)/params/gazebo/global_planner_params.yaml" command="load"/>
         <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
         <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
         <param name="local_costmap/global_frame" value="$(arg global_frame_id)"/>
         <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
         <param name="DWAPlannerROS/global_frame_id" value="$(arg global_frame_id)"/>
         <remap from="cmd_vel" to="$(arg cmd_topic)"/>
         <remap from="odom" to="$(arg odom_topic)"/>
         <remap from="scan" to="$(arg laser_topic)"/>
         <remap from="map"  to="$(arg map_topic)"/>
      </node>
   </group>
</launch>